src/Display/Slider/Slider.test.tsx

```diff
@@ -0,0 +1,311 @@
+          document.querySelectorAll('li')[index].classList.contains('active')
+        ).toBeTruthy();
+      });
+
+      // mock getSliderContainerDOMNode to calculate translateX
+      sliderRef.current.getSliderContainerDOMNode = () => ({
+        getBoundingClientRect: () => ({
+          width: 500,
+        }),
+      });
+
+      rerender(
+        <Slider ref={sliderRef} initialItem={initialItem}>
+          <Slider.Item>a</Slider.Item>
+          <Slider.Item>b</Slider.Item>
+          <Slider.Item>c</Slider.Item>
+        </Slider>
+      );
+      expect(asFragment()).toMatchSnapshot();
+    });
+  };
+
+  [0, 1, 2, 3].forEach(initialItem => matchSnapshot(initialItem));
+});
+
+describe('<Slider/> prop fullContent', () => {
+  const matchSnapshot = (fullContent: boolean) => {
+    it(`should match snapshot when fullContent ${fullContent} is passed`, async () => {
+      const { asFragment } = render(
+        <SliderComponent fullContent={fullContent} />
+      );
+      expect(asFragment()).toMatchSnapshot();
+    });
+  };
+
+  [true, false].forEach(fullContent => matchSnapshot(fullContent));
+});
+
+describe('<Slider/> prop arrowWhite', () => {
+  const matchSnapshot = (arrowWhite: boolean) => {
+    it(`should match snapshot when arrowWhite ${arrowWhite} is passed`, async () => {
+      const { asFragment } = render(
+        <SliderComponent arrowWhite={arrowWhite} />
+      );
+      expect(asFragment()).toMatchSnapshot();
+    });
+  };
+
+  [true, false].forEach(arrowWhite => matchSnapshot(arrowWhite));
+});
+
+describe('<Slider/> prop removeDots', () => {
+  const matchSnapshot = (removeDots: boolean) => {
+    it(`should match snapshot when removeDots ${removeDots} is passed`, async () => {
+      const { asFragment } = render(
+        <SliderComponent removeDots={removeDots} />
+      );
+      expect(asFragment()).toMatchSnapshot();
+    });
+  };
+
+  [true, false].forEach(removeDots => matchSnapshot(removeDots));
+});
+
+describe('<Slider/> prop autoplay', () => {
+  it(`should go to next slide after 6000ms when autoplay is truthy`, () => {
+    jest.useFakeTimers();
+    const { asFragment } = render(<SliderComponent autoplay={true} />);
+
+    // validate initial active slide
+    expect(
+      document.querySelectorAll('li')[0].classList.contains('active')
+    ).toBeTruthy();
+    expect(asFragment()).toMatchSnapshot();
+
+    // validate autoplay
+    jest.advanceTimersByTime(6000);
+    expect(
+      document.querySelectorAll('li')[1].classList.contains('active')
+    ).toBeTruthy();
+    expect(asFragment()).toMatchSnapshot();
+  });
+
+  it(`should go to the first slide after 6000ms when autoplay is truthy and the current slide is the last slide`, () => {
+    jest.useFakeTimers();
+    const { asFragment } = render(
+      <SliderComponent autoplay={true} initialItem={3} />
+    );
+
+    // validate initial active slide
+    expect(
+      document.querySelectorAll('li')[2].classList.contains('active')
```

Comment:
Would it be possible to have a `const listItems = document.querySelectorAll('li')` at the top of the test case and then do `listItems[0]`? And `listItems[listItems.length - 1]` when we're targeting the last item, to make it clearer that it's the last?

Reply:
ok I will name the li array.